src/pages/Messages.jsx

```diff
@@ -1,24 +1,37 @@
-import React, { useEffect } from "react";
+import React, { useState, useEffect } from "react";
 import { useParams } from "react-router-dom";
+//components
 import Sidebar from "@features/sidebar/components/Sidebar";
-import MessageList from '../features/messages/components/MessageList'
+import MessageList from "../features/messages/components/MessageList";
 import HeroSection from "../features/heroSection/components/HeroSection";
-
+// api url
+import { MESSAGES_URL } from "@api/apiConstants.js";
 const Messages = () => {
-  let { categoryName } = useParams();
-
+  let { categoryId } = useParams();
+  const [messages, setMessages] = useState(null);
   //TODO: fetch messages based on the "categoryName" to the API
   useEffect(() => {
-    console.log(`fetching ${categoryName} messages!`);
-  }, [categoryName]);
+    //! demo fetch, to be used only as demo, replace with react query.
+    const getData = async () => {
+      if (!categoryId) {
+        categoryId = "all";
+      }
+      const response = await fetch(`${MESSAGES_URL}/category/${categoryId}`);
```

Comment:
this path should be changed , use query instead of params  ( u can look at the backend comment i gave u on this route)

Reply:
even if I only send categoryIDs?